src/common/utils/resolve_reference.py

```diff
@@ -237,8 +246,10 @@ class ResolvedReference:
     entity: dict | list
 
 
-def resolve_reference(reference: str, get_data_source: Callable) -> ResolvedReference:
-    """Resolve the reference into a document."""
+def resolve_reference(reference: str, get_data_source: Callable, resolve: bool = False) -> ResolvedReference:
+    """Resolve the reference into a document.
+
+    resolve: Resolve the target if it's a reference"""
```

Comment:
The extra "resolve" argument is q bit confusing. 
I'm not sure what it means to call resolve_reference() with resolve = False? That document returned in the  ResolvedReference is not a dict?